test/suites/nodeclaim/nodeclaim_test.go

```diff
@@ -377,4 +377,45 @@ var _ = Describe("StandaloneNodeClaim", func() {
 		// Expect that the nodeClaim is eventually de-provisioned due to the registration timeout
 		env.EventuallyExpectNotFoundAssertion(nodeClaim).WithTimeout(time.Minute * 20).Should(Succeed())
 	})
+	It("should wait for instance termination before removing finalizer from nodeClaim", func() {
+		nodeClaim := test.NodeClaim(corev1beta1.NodeClaim{
+			Spec: corev1beta1.NodeClaimSpec{
+				Requirements: []corev1beta1.NodeSelectorRequirementWithMinValues{
+					{
+						NodeSelectorRequirement: v1.NodeSelectorRequirement{
+							Key:      v1beta1.LabelInstanceCategory,
+							Operator: v1.NodeSelectorOpIn,
+							Values:   []string{"c"},
+						},
+					},
+					{
+						NodeSelectorRequirement: v1.NodeSelectorRequirement{
+							Key:      corev1beta1.CapacityTypeLabelKey,
+							Operator: v1.NodeSelectorOpIn,
+							Values:   []string{corev1beta1.CapacityTypeOnDemand},
+						},
+					},
+				},
+				NodeClassRef: &corev1beta1.NodeClassReference{
+					Name: nodeClass.Name,
+				},
+			},
+		})
+		env.ExpectCreated(nodeClass, nodeClaim)
+		node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+		nodeClaim = env.EventuallyExpectCreatedNodeClaimCount("==", 1)[0]
+
+		instanceID := env.ExpectParsedProviderID(node.Spec.ProviderID)
+		env.GetInstance(node.Name)
+
+		env.ExpectDeleted(node)
```

Comment:
Why are we deleting the node here? Shouldnt we delete the NodeClaim?

Reply:
Made the change.